lib/redbird/plug/session/redis.ex

```diff
@@ -13,26 +15,30 @@ defmodule Plug.Session.REDIS do
     opts
   end
 
-  def get(_conn, namespaced_key, _init_options) do
-    case get(namespaced_key) do
-      :undefined -> {nil, %{}}
-      value -> {namespaced_key, value |> :erlang.binary_to_term()}
+  def get(conn, prospective_key, _init_options) do
+    with {:ok, key, _} <- Key.extract_key(prospective_key),
+         {:ok, _verified_key} <- Key.verify(key, conn),
+         value when is_binary(value) <- get(prospective_key) do
+      {prospective_key, Crypto.safe_binary_to_term(value)}
+    else
+      _ -> {nil, %{}}
     end
   end
 
   def put(conn, nil, data, init_options) do
-    put(conn, add_namespace(generate_random_key()), data, init_options)
+    put(conn, Key.generate(), data, init_options)
   end
 
-  def put(_conn, namespaced_key, data, init_options) do
-    value = :erlang.term_to_binary(data)
+  def put(conn, key, data, init_options) do
+    key
+    |> Key.sign_key(conn)
+    |> set_key_with_retries(:erlang.term_to_binary(data), session_expiration(init_options), 1)
```

Comment:
Along with my comment about `Redbird.Key` being the only module that depends on `Redbird.Crypto`, what do you think about moving this `:erlang.term_to_binary/1` call into `Redbird.Key`? 

Perhaps something like `Redbird.Key.encode/1` would capture what we're doing here?

Reply:
Similar to my comment above, this is dealing with the value and not the key.